src/Microsoft.AspNet.OData.Shared/Formatter/ODataModelBinderConverter.cs

```diff
@@ -30,8 +30,20 @@ namespace Microsoft.AspNet.OData.Formatter
         Justification = "Relies on many ODataLib classes.")]
     public static class ODataModelBinderConverter
     {
-        private static readonly MethodInfo EnumTryParseMethod = typeof(Enum).GetMethods()
+#if NETSTANDARD2_0
+        internal static readonly MethodInfo EnumTryParseMethod = typeof(Enum).GetMethods()
             .Single(m => m.Name == "TryParse" && m.GetParameters().Length == 2);
+#else
+        // .NET 6 adds a new overload: TryParse<TEnum>(ReadOnlySpan<Char>, TEnum)
+        // Now, with `TryParse<TEnum>(String, TEnum)`, there will have two versions with two parameters
+        // So, the previous Single() will throw exception.
+        internal static readonly MethodInfo EnumTryParseMethod = typeof(Enum).GetMethod("TryParse",
+            new[]
+            {
+                typeof(string),
+                Type.MakeGenericMethodParameter(0).MakeByRefType()
+            });
```

Comment:
Would there be a problem if you used this method regardless of the .net version, i.e. (without the `#if NETSTANDARD2_0`)?

Reply:
Actually, I've seen that it uses methods not supported in netstandard2.0